controllers/webhook/webhook.go

```diff
@@ -122,3 +124,46 @@ func (w *OpenDataHubWebhook) Handle(ctx context.Context, req admission.Request)
 
 	return admission.Allowed(fmt.Sprintf("Operation %s on %s allowed", req.Operation, req.Kind.Kind))
 }
+
+//+kubebuilder:webhook:path=/mutate-opendatahub-io-v1,mutating=true,failurePolicy=fail,sideEffects=None,groups=datasciencecluster.opendatahub.io,resources=datascienceclusters,verbs=create;update,versions=v1,name=mutate.operator.opendatahub.io,admissionReviewVersions=v1
+//nolint:lll
+
+// OpenDataHubMutatingWebhook is a mutating webhook
+// It currently only sets defaults for modelregiestry in datascienceclusters.
+type OpenDataHubMutatingWebhook struct {
+	Client  client.Client
+	Decoder *admission.Decoder
+}
+
+func (w *OpenDataHubMutatingWebhook) SetupWithManager(mgr ctrl.Manager) {
+	hookServer := mgr.GetWebhookServer()
+	odhWebhook := &webhook.Admission{
+		Handler: w,
+	}
+	hookServer.Register("/mutate-opendatahub-io-v1", odhWebhook)
```

Comment:
i think the main reason to use 2 differnet endpoints , one per each, instead of having one endpoint for both (which is applicable as well)  is we already had validation webhook in downstream. By this way, we can get the downstream part out first, and the mutating part as a separate one to test. 
another reason is, it is slightly easy to maintain, the logic will be simple in its own Handle() and mutating only need on DSC not DSCI  for now.

Reply:
> According to the k8s documentation:
> 
> ```
> When allowing a request, a mutating admission webhook may optionally modify the incoming object as well. 
> ```
> 
> So, we need a mutating webhook to modify a CR. They also use different sets of CRs and verbs.

Do I understand right that you claim that it is impossible?

Reply:
How are you reading the documentation? Please point me to the documentation where it says a validating webhook can respond with a patch, maybe I missed it.

Reply:
Have you considered using [`admission.CustomDefaulter`](https://github.com/kubernetes-sigs/controller-runtime/blob/2eb879f25c4829825e6f2511cda035de41ec7030/pkg/webhook/admission/defaulter_custom.go) instead? It seems like a good fit if we want to follow the single responsibility principle. It would also simplify the code.

Reply:
> How are you reading the documentation? Please point me to the documentation where it says a validating webhook can respond with a patch, maybe I missed it.

I'm just asking questions :)

Reply:
> i think the main reason to use 2 differnet endpoints , one per each, instead of having one endpoint for both (which is applicable as well) is we already had validation webhook in downstream. By this way, we can get the downstream part out first, and the mutating part as a separate one to test. another reason is, it is slightly easy to maintain, the logic will be simple in its own Handle() and mutating only need on DSC not DSCI for now.

I did some experiments and agree, it makes logic more simple.
But I have another question. It has a check for dsc.Spec.Components.ModelRegistry.RegistriesNamespace but with // +kubebuilder:default="odh-model-registries" it never true for me.

Reply:
I considered `CustomDefaulter`, but it doesn't pass along the previous version of the object. It's nice to have it if the defaulting logic ever becomes complex enough to know what the old spec was. 
We are already using complex validation logic in the MR CEL validator. So, I don't know whether defaulting webhook will ever need to also do that, but I also don't know that it never will. :smile: